coders/xcf.c

```diff
@@ -1081,16 +1170,28 @@ static Image *ReadXCFImage(const ImageInfo *image_info,ExceptionInfo *exception)
       image=DestroyImageList(image);
       return((Image *) NULL);
     }
-  count=ReadBlob(image,14,(unsigned char *) magick);
-  if ((count != 14) ||
+  count=ReadBlob(image,sizeof(magick),(unsigned char *) magick);
+  if ((count != sizeof(magick)) ||
       (LocaleNCompare((char *) magick,"gimp xcf",8) != 0))
     ThrowReaderException(CorruptImageError,"ImproperImageHeader");
   (void) memset(&doc_info,0,sizeof(XCFDocInfo));
+  count=ReadBlob(image,sizeof(version),(unsigned char *) version);
+  if (count != sizeof(version))
+    ThrowReaderException(CorruptImageError,"ImproperImageHeader");
+  doc_info.version=strtoul(version, NULL, 10);
```

Comment:
Sorry that I missed that you are reading a string here instead. We could revert back to `magick[14]` here then but keep the `sizeof`. But could you use `StringToUnsignedLong` instead of `strtoul`?